private-templates-service/adaptivecards-templating-service/src/storageproviders/MongoDBProvider.ts

```diff
@@ -0,0 +1,208 @@
+import mongoose from "mongoose";
+import { StorageProvider } from "./IStorageProvider";
+import { IUser, ITemplate, JSONResponse } from "../models/models";
+import { IUserModel, UserSchema } from "../models/mongo/UserModel";
+import { ITemplateModel, TemplateSchema } from "../models/mongo/TemplateModel";
+import { Mongo } from "../util/mongoutils";
+
+const USERS_COLLECTION_NAME_SINGULAR: string = "User";
+const TEMPLATES_COLLECTION_NAME_SINGULAR: string = "Template";
+
+export class MongoDBProvider implements StorageProvider {
+  Template!: mongoose.Model<ITemplateModel>;
+  User!: mongoose.Model<IUserModel>;
+  db!: mongoose.Connection;
+  connectionString: string;
+  options: any;
+  async getUser(query: Partial<IUser>): Promise<JSONResponse<IUser[]>> {
+    return await this.User.find(query)
+      .then(users => {
+        if (users.length) {
```

Comment:
Is there supposed to be a condition here? ( .length > 0)?

Reply:
if length is 0, it will be evaluated to false anyway :)

Reply:
if you don't need to check length, then just `if (users)` ?

Reply:
becayse if mongo doesn't find anything that matches the query - it will return emty array `[]` and for some weird reasons in typescript empty array passes if statement: `if([]) => true`
so I have to check for length :)

Reply:
oh yeah, okay!